### ISSUE TITLE
Volunteers welcome: Compare mame2003+ bootstrap list to mainline game FAQ

### ISSUE THREAD

**Issue by markwkidd** [libretro/mame2003-plus-libretro#1084]
At this point the bootstrap collection has grown to be quite extensive. To wit, this is the list of bootstraps:
https://github.com/libretro/mame2003-plus-libretro/blob/master/src/bootstrap.h

@KMFDManic has pointed out that mainline maintains a list of games that need some kind of configuration to work as expected. Many of these are the same games that have been bootstrapped. Here's that list: https://wiki.mamedev.org/index.php/FAQ:Games

Any interested parties out there want to see if we are missing any potential bootstraps?

**Comment by mahoneyt944** [comment]
Listed

- 2 on 2 Open Ice Challenge
- 3 Bags Full
- 720 Degrees
- American Poker 95
- American Poker II (and bootlegs)
- Amuse
- Animal Catch
- Apple 10
- Arm Champs II
- Asterix
- Autumn Moon
- Avengers in Galactic Storm
- Beast Busters
- Bells & Whistles
- Big Deal
- Big Karnak
- Black Rhino
- Blaster
- Bosconian
- Bottle 10
- Bubble Memories
- Bubble Trouble
- Bubbles
- Bucky O'Hare
- Captain Commando
- CarnEvil
- Championship Bowling
- Champion Super
- Complex X
- Coral Riches II
- Crazy Monkey
- Critter Crush
- Cuore 1
- Cyber Cycles
- Cyber Sled
- Dark Horse (Bootleg)
- Dark Horse Legend
- Daytona USA
- Defender (and bootlegs)
- Detana!! Twin Bee
- Dirt Fox
- Double Dragon
- Dragoon Might
- Dream Soccer '94
- Dungeons & Dragons: Shadow over Mystara
- Elephant Family
- Enchanted Forest
- Escape Kids
- Fantastic Journey
- Fantasy Fortune
- Final Lap 2
- Final Lap 3
- Final Lap R
- Find Love
- Five-A-Side Soccer
- Food Fight
- Four Roses
- Fruit Cocktail
- Funky Head Boxers
- Gaiapolis
- Garage
- GI Joe
- Gokujyou Parodius
- Golden Canaries
- Golden Tee Fore series
- Golfing Greats 2
- Golly Ghost
- GTI Club
- Hae Hae Ka Ka
- Hang Pilot
- Hard Drivin' / Race Drivin'
- Hyper Sports
- Il Pagliaccio
- Inferno
- Island / Island 2
- Itazura Monkey
- Jockey Club
- Jolli Witch
- Jolly Card
- Jolly Joker
- JoJo's Bizarre Adventure / Venture
- Joust
- Joust 2
- K.G Bird
- Karate Blazers
- Keks
- King of Fighters (Series)
- King Pin / King Pin Multi-Game
- Konami 80's AC Special
- Konami's Open Golf Championship
- Kram
- Kyukyoku Sentai Dadandarn
- Laser Ghost
- Lethal Enforcers
- Lethal Enforcers 2
- Line of Fire
- Lotto Fun
- Lovely Poker
- Lucky & Wild
- Lucky Haunter
- Magic's 10
- Magic Card II
- Magical Zunou Power
- Mahjong Pon Chin Kan
- Mahjong Yuugi
- Marchen Maze
- Martial Champion
- Metal Slug (Series)
- Metamorphic Force
- Microman Battle Charge
- Millennium Nuovo 4000
- Monster Maulers
- Mortal Kombat II / Mortal Kombat 3 / Ultimate Mortal Kombat 3
- Multi Fish
- Mystic Marathon
- Mystic Warriors
- Name Club Ver.3
- Narc
- NBA Hangtime / NBA Maximum Hangtime
- Pac-Slot
- Phantom Pays
- Phrase Craze
- Pirate
- Pirate Ship
- PlayChoice-10
- Player's Edge Plus (Set038) Set Chip
peset038
- Player's Edge Plus games (peplus.cpp)
- Poker - Greyhound Electronics
- Poker 4-1
- Poker De Win
- Poker Master (Tony Poker) Hack
- Pontoon
- Pool 10
- Pot Game
- Premier Soccer
- Print Club 2
- Pro Mahjong Kiwame
- Pro Mahjong Kiwame S
- Prop Cycle
- Pull Tabs
- Pye-nage Taikai
- Qix
- Quiz Gakumon no Susume
- Quiz Sangokushi
- Quiz Tonosama no Yabou
- Racing Force
- Raiden Fighters Jet (US, Single Board)
- Rampage World Tour
- Red Earth
- Resident
- Revolution X
- Rim Rockin' Basketball
- Robotron
- Rock Climber (rclimb_3)
- Roll Fruit
- Royal Card
- Royal Vegas Joker Card
- Run and Gun
- Salamander 2
- San Fransisco Rush
- Scud Race
- Sea Bass Fishing
- Selection - Greyhound Electronics
- Sengoku Mahjong [BET]
- Shadow Dancer
- Sigma Poker 2000
- Sigma Poker
- Simpsons Bowling
- Sinistar
- Slam Dunk
- Slither
- Snooker 10
- Soccer Superstars
- Solar Assault
- Space Dungeon
- Space Gun
- Speed Racer
- Splat
- Stargate
- Steel Gunner / Steel Gunner 2
- Steep Slope Sliders
- Street Fighter III (Series)
- Sunset Riders
- Super Gran Safari
- Super High Impact
- Super Pool 99
- Super Real Mahjong 2
- Sweet Hearts II
- Sweet Life
- Taihou de Doboon
- Taisen Puzzle-dama
- Taisen Tokkar-dama
- Target Panic
- Technical Bowling
- Techno Drive
- Teenage Mutant Ninja (Hero) Turtles 2 - Turtles in Time
- Ten Balls
- Terminator 2
- The Couples
- The Electric Yo-Yo
- The Simpsons
- Thunder Cross II
- Time Crisis
- Tobikose! Jumpman
- Tokimeki Memorial Taisen Puzzle-dama
- Top Gear
- Top Gunner/Vertigo
- Tortuga Family
- Track & Field
- Triple Hunt
- Turkey Shoot
- Twin Bee Yahhoo!
- Vendetta
- Videomat
- Violent Storm
- Virtual Mahjong
- Virtual Mahjong 2 - My Fair Lady
- Warzard
- Water Match
- White Tiger
- Wild West C.O.W.-Boys of Moo Mesa
- Witch
- World Tennis
- WWF Wrestlemania
- X-Men
- Xexex
- Zero Team 2000 (zerotm2k)
- Zombie Raid
- Zoo Keeper

**Comment by mahoneyt944** [comment]
Updated the above post, removed all games we do not support in this core and games we have already bootstrapped. Next step is to remove any game known to not be working due to driver issues. This should make the check list fairly reasonable, more so than it already is :)

**Comment by KMFDManic** [comment]
@mahoneyt944 @arcadez2003 @grant2258 @markwkidd A game that is clearly not playable on slower platforms (Killer Instinct!), I was able to get working fairly well.  Once I analyze things further, it might be a good game to "bootstrap" based on the necessary math involved to run it like in my video.  The only problem is that we may need to add a Core Option to pull this off, as it may break other Platforms!  I am still testing and seeing how to go about this, so it doesn't really screw up platforms that can already run it well.

https://www.youtube.com/watch?v=Au1i0AKmFes

In contrast, here is how we USED to run the KI 1 and 2 games!  At the 7:06 time stamp!

https://www.youtube.com/watch?v=1L3txhiSTgc&t=1s

**Comment by Darknior** [comment]
> @mahoneyt944 @arcadez2003 @grant2258 @markwkidd A game that is clearly not playable on slower platforms (Killer Instinct!), I was able to get working fairly well. Once I analyze things further, it might be a good game to "bootstrap" based on the necessary math involved to run it like in my video. The only problem is that we may need to add a Core Option to pull this off, as it may break other Platforms! I am still testing and seeing how to go about this, so it doesn't really screw up platforms that can already run it well.

WOAW !!! It's incredible to see what you do, the game is really better with your hack !!

**Comment by KMFDManic** [comment]
@mahoneyt944 one thing that would probably be a good thing to do, now, is bootstrap higher volume for Killer Instinct.  The game runs SOOOO much better now, and better volume will be bootstrap icing on the cake.

  Other than that, Cruis'n World would be benefit from overall control nvram bootstrapping, as well as volume boost.  I believe cruisn usa may still possibly need volume increase, too?

I will try out NFL Blitz next!

**Comment by mahoneyt944** [comment]
I'll have to poke around with those bootstraps.

**Comment by KMFDManic** [comment]
@mahoneyt944 note, if messing with clock scale, while testing Cruisn Games, make sure it is at 0, or the boot sequence may keep restarting.  But, once in game, you should gain better performance the same way you did with Blitz.  World is quite playable for me now, albeit, we just need controls and volume to better handle things on the end user part.  I will keep you posted on anything else.  But, the above mentioned games, including KI will be tremendous in of themselves.  I'm also curious how well KI and World run for you.  Be sure to let me know.

**Comment by KMFDManic** [comment]
@mahoneyt944 Here is my personal WIP of Cruis'n World.  Controls still need a little touch up.  Sound is much nicer, louder.  Performance/Speed are exponentially than in my earlier videos where I showcased the game/s of USA/World running like an actual slide show!

https://www.youtube.com/watch?v=9xSg0eQ051o

**Comment by KMFDManic** [comment]
@grant2258 @arcadez2003 I accidentally posted before pinging you two.  You may appreciate the old/new difference in performance/speed, as well!  Any games, either of you, as well as mahoneyt944 have trouble running well, I definitely don't mind trying to help iron out kinks with!  Shame we can't do Outfoxies with sound:)  But, we have plenty of other stuff to work with!  So, no complaints:)

**Comment by mahoneyt944** [comment]
Well I use a cellular Android platform and it runs just about everything full speed on this core... that I've tried anyhow, besides some of the chd titles of course. Those typically play at maybe 25 fps and I can get a 15 - 20 fps boost with clock scaling and make them more playable. The sweet spot seems to be a setting low enough to boost the fps but high enough to not have slow motion glitchy graphics.

**Comment by KMFDManic** [comment]
@mahoneyt944 Also, I am a fan of MK.  I really need to get the new one with Rambo in it!  But, UMK3 was awesome, especially with the re-add of Kitana.  I can certainly do a demonstration, if you want me to.  Let me know what stands out as needing showcased.  And, the easiest way for me to grab stuff would be if you do a "temp link" on any hosting site, even if its google drive "public shared", temporarily.  If you respond to this message with the link, and then delete the link 1 minute later, I will "still" get the message before you edited it.  We can go from there:)  Thanks in advance for checking into any of this stuff.  MK is a beautiful thing, too!

Be sure to check out the animated WB movie, Scorpion's Revenge, Part 1.  part 2 is coming out soon, I believe.  And, I did check out the latest Live Action MK, as well.  Some didn't like the new character.  But, overall, being a MK fan, I enjoyed the flick for the popcorn entertainment it was meant to be!

**Comment by mahoneyt944** [comment]
@KMFDManic well theres lots of extras to many to count. 
- the selection screen has human smoke, noob saibot, added. Fully playable.
- Rain is available but hidden to select. Hover over reptile and press "up + start + high kick" at the same time. You'll hear an explosion sound indicating it selected rain.
- The 3 unlockable characters are available by default.
- on the towers selection screen press start to toggle to all endurance mode or regular.
- lots of music corrections
- There's a whole umk3+ menu in the games service menu to enable custom start screens, level preferences, etc.
- there's also special modes that can get activated while playing threw a tower like silent Kombat, super uppercuts, dark Kombat. 

Here's a temp link 
[umk3p.zip](https://github.com/libretro/mame2003-plus-libretro/files/6866030/umk3p.zip)

**Comment by KMFDManic** [comment]
@mahoneyt944 Thanks for the link.  I had to work back to back OT 3 days in a row.  So, didn't get much time to pop in:)  But, should be able to get more stuff done now!  I will check out the +:)

**Comment by KMFDManic** [comment]
@mahoneyt944 If you have a moment to humor me.  I have been playing more around with some of the bootstraps you've added into 2003 Plus.  In the event, I would like to create my own patches, so nvrams can auto generate, could you do a brief run through?  I assume bin2c was used?  If I can get any useful and meaningful bootstraps together, I will let you know, as well as potentially pull request if they are stable enough for 2003 Plus:)  Definitely quite a few games left to get these done for.  So, why not!:)

**Comment by mahoneyt944** [comment]
The version we use here pretty much does all the leg work. You just manually copy the size back to the top of the declaration. Have you used bin2c yet?

**Comment by KMFDManic** [comment]
> The version we use here pretty much does all the leg work. You just manually copy the size back to the top of the declaration. Have you used bin2c yet?

No, have not personally used it.  Just wanted to hopefully have an example perimeter to ensure I do things correctly.  I'd assume I'd start with a manual nvram .nv file and then go from there as far as converting it to a c file for 2003 plus codeline?

**Comment by mahoneyt944** [comment]
Yeah. I use to run it on my pi, I now have a Windows version of it which uses it as a batch file for ease of use.

But it's really that simple, make a nv file you wish to hard code, specify this file to bin2c. Run it, it will spit out a text file with the results in a format ready to copy to the core. You'll notice the size is listed at the end, so you do have to copy that back to the top. Do you run Linux nor Windows or?

**Comment by KMFDManic** [comment]
@mahoneyt944 I have ubuntu Linux as well as Windows 10.  I didn't think to try to use a Windows Version, since I just started poking around with it.  I just think it would be a lot more convenient, not to mention nice, to help add to the repertoire of already existing bootstraps.  If you have a batch file handy, and link to the bin2c one you are using for Windows, that would be fantastic:)

**Comment by mahoneyt944** [comment]
Sure. I'm not home right now to send it but I can probably get it to you in the morning.

**Comment by KMFDManic** [comment]
> Sure. I'm not home right now to send it but I can probably get it to you in the morning.

Much appreciated:)

**Comment by KMFDManic** [comment]
Thanks, @mahoneyt944 I will grab and test this stuff out once home today.  I just need to NULL the 2nd to last part of the GAMEC part to make it work with 2003 Xtreme.

I will let you know if I have any other questions.  Once I get things going, I will keep you updated if I get any worthy bootstraps together for Plus.  Thanks again!

**Comment by KMFDManic** [comment]
@mahoneyt944 @grant2258 

I feel we should try to bootstrap hard drivin' into 2003 plus.  I tested mahoney's nvram and it controlled great.  The only odd thing is in order to shift into drive, I had to go to mame menu and remap the shifter that is binded to f or down, to be my gas.  Then, I had no issues controlling it:)

**test video with hard drivin'**

https://youtu.be/0Dwd3DjQTb8?si=sw--SLzK04lDU4qQ

Thoughts?

**Comment by mahoneyt944** [comment]
The bootstrap is trivial, I just think we need to hack the analog stick and make it more natural. It's hard to play without a fixed position shifter.

**Comment by KMFDManic** [comment]
> The bootstrap is trivial, I just think we need to hack the analog stick and make it more natural. It's hard to play without a fixed position shifter. 

Any ideas of what we could do to even pull off the fixed shifter? And, I definitely get you on that aspect.  It is a bit irritating that such a cool game is mired in control issues.

**Comment by mahoneyt944** [comment]
I'd probably set it up like a modern racer, with 2 buttons for up and down shifting and maybe one to just to neutral

**Comment by KMFDManic** [comment]
@mahoneyt944 Thanks for the send of both the hard drivin' nvram, as well as the boostrap stuff.  Everything great on my end:)  If you feel you can pull off the method of "modern racer" set-up, be sure to let me know.  Otherwise, in the meantime, I am gonna play around with a few bootstraps, as well as test Hard Drivin' further:)  Thanks again!  Plus/Xtreme FTW:)

**Comment by KMFDManic** [comment]
@mahoneyt944 I did nvram bootstraps for Metal Slug 1 and 2, to have blood auto enabled.  The neogeo.c driver uses GAMEB, instead of GAME.  And, it generates conflict errors as a result, it seems, due to trying to change to GAMEC.  Curious as to your thoughts on this one and how to handle it.  I am starting to get other nvrams done.  But, figured, Metal Slug 1 and 2, with blood activated, MUST HAVE!  for now:)

[mslug_1_&_2_nvram.zip](https://github.com/libretro/mame2003-plus-libretro/files/14184243/mslug_1_._2_nvram.zip)

**Comment by mahoneyt944** [comment]
I don't believe we have that functionality for bios required games as you've found. You'd have to make a whole new macro to hand that.

**Comment by KMFDManic** [comment]
@mahoneyt944 I kind of felt tackling a neogeo game might require a little additional coding.  Would have been nice if it could be a near copy and paste like GAMEC.  Will still keep this on the agenda, however:)

**Comment by wn2000** [comment]
`sshooter` needs a bootstrap (nvram) to calibrate the gun.
